cli/src/helpers/installDependencies.ts

```diff
@@ -1,27 +1,78 @@
 import chalk from "chalk";
 import { execa } from "execa";
-import ora from "ora";
-import { getUserPkgManager } from "~/utils/getUserPkgManager.js";
+import ora, { Ora } from "ora";
+import {
+  getUserPkgManager,
+  PackageManager,
+} from "~/utils/getUserPkgManager.js";
 import { logger } from "~/utils/logger.js";
 
 type Options = {
   projectDir: string;
 };
 
+/*eslint-disable @typescript-eslint/no-floating-promises*/
+const runInstallCommand = async (
+  pkgManager: PackageManager,
+  projectDir: string,
+): Promise<Ora | null> => {
+  switch (pkgManager) {
+    // When using npm, inherit the stderr stream so that the progress bar is shown
+    case "npm":
+      await execa(pkgManager, ["install"], {
+        cwd: projectDir,
+        stderr: "inherit",
+      });
+
+      return null;
+    // When using yarn or pnpm, use the stdout stream and ora spinner to show the progress
+    case "pnpm":
+      const pnpmSpinner = ora("Running pnpm install...").start();
+      const pnpmSubprocess = execa(pkgManager, ["install"], {
+        cwd: projectDir,
+        stdout: "pipe",
+      });
+
+      await new Promise<void>((res, rej) => {
+        pnpmSubprocess.stdout?.on("data", (data: Buffer) => {
+          if (data.toString().includes("Progress")) {
```

Comment:
```suggestion
          const text = data.toString();
          if (text.includes("Progress") && text.includes("|")) {
```

Seems like not every pnpm version uses the `|` so we should check for that too:

![CleanShot 2023-01-28 at 12 19 32](https://user-images.githubusercontent.com/51714798/215263747-548ba18a-6302-44c6-ae21-49fc31dadad4.png)

Reply:
https://user-images.githubusercontent.com/51714798/215263845-fd219e4a-1988-44ab-8298-80fba602d0ab.mp4

I for example doesn't get any of the progress output here

Reply:
I'm still checking for the progress text for older versions, no reason to show lines like `virtual store is at:` imo. Thanks for the info tho, I've updated the pr :+1:

Reply:
![CleanShot 2023-01-28 at 12 26 10](https://user-images.githubusercontent.com/51714798/215263988-b6db8f60-1bc5-4520-bb18-2da5556880ea.png)

nice